emcc.py

```diff
@@ -462,7 +462,11 @@ def is_supported(f):
         # lld allows various flags to have either a single -foo or double --foo
         if f.startswith(flag) or f.startswith('-' + flag):
           diagnostics.warning('linkflags', 'ignoring unsupported linker flag: `%s`', f)
-          return False, takes_arg
+          # Skip the next argument if this linker flag takes and argument and that
+          # argument was not specified as a separately (i.e. it was specified as
+          # single arg containing an `=` char.)
```

Comment:
```suggestion
          # Skip the next argument if this linker flag takes an argument and that
          # argument was not specified separately (i.e. it was specified as
          # single arg containing an `=` char.)
```
Is that right?

If so then this makes sense to me. But then isn't this a separate fix from the PR?

Reply:
Ah yes, this is a bug fix, but it also happens to be needed as part of this change.  I split it out as #14356.

The reason this change depends on it is that other wise `test_supported_linker_flags` will start failing.   The reason this change depends on that fix goes like this:

1. We specify the `-L/path/to/sysroot` as the first system link flag returned by `emsdk_ldflags`
2. System link flags get added to user link flags
3. If, as in that test, that last user link flag is an ignored one of the form `-Wl,-rpath=foo` then the next argument will be ignored.. which will be the `-L/path/to/sysroot`
4. This will cause `-lc` to not be found, but that didn't matter previously because we were using full paths the libc.a prior to this change.